python/lsst/obs/base/fitsExposureFormatter.py

```diff
@@ -219,5 +263,133 @@ def write(self, inMemoryDataset):
         """
         # Update the location with the formatter-preferred file extension
         self.fileDescriptor.location.updateExtension(self.extension)
-        inMemoryDataset.writeFits(self.fileDescriptor.location.path)
+        outputPath = self.fileDescriptor.location.path
+
+        # check to see if we have a recipe requested
+        recipeName = self.writeParameters.get("recipe")
+        recipe = self.getImageCompressionSettings(recipeName)
+        if recipe:
+            # Can not construct a PropertySet from a hierarchical
+            # dict but can update one.
+            ps = PropertySet()
+            ps.update(recipe)
+            inMemoryDataset.writeFitsWithOptions(outputPath, options=ps)
+        else:
+            inMemoryDataset.writeFits(outputPath)
         return self.fileDescriptor.location.pathInStore
+
+    def getImageCompressionSettings(self, recipeName):
+        """Retrieve the relevant compression settings for this recipe.
+
+        Parameters
+        ----------
+        recipeName : `str`
+            Label associated with the collection of compression parameters
+            to select.
+
+        Returns
+        -------
+        settings : `dict`
+            The selected settings.
+        """
+        # if no recipe has been provided and there is no default
+        # return immediately
+        if not recipeName:
+            if "default" not in self.writeRecipes:
+                return {}
+            recipeName = "default"
+
+        if recipeName not in self.writeRecipes:
+            raise RuntimeError(f"Unrecognized recipe option given for compression: {recipeName}")
+
+        recipe = self.writeRecipes[recipeName]
+
+        # Set the seed based on dataId
+        seed = hash(tuple(self.dataId.items())) % 2**31
+        for plane in ("image", "mask", "variance"):
+            if plane in recipe and "scaling" in recipe[plane]:
+                scaling = recipe[plane]["scaling"]
+                if "seed" in scaling and scaling["seed"] == 0:
+                    scaling["seed"] = seed
+
+        return recipe
+
+    # @staticmethod
+    def validateWriteRecipes(self, recipes):
```

Comment:
@jonathansick points out that pydantic would be perfect here instead of hand crafted validation.